kodilla-good-patterns/src/main/java/com/kodilla/good/patterns/food/ProductOrderService.java

```diff
@@ -0,0 +1,30 @@
+package com.kodilla.good.patterns.food;
+
+import com.kodilla.good.patterns.food.product.Product;
+import com.kodilla.good.patterns.food.supplier.Supplier;
+
+import java.util.Optional;
+
+public class ProductOrderService implements OrderService {
+
+    @Override
+    public boolean order(OrderRequest orderRequest, Optional<Product> product, Integer amount, Supplier supplier) {
+        return product.map(value -> isPresent(orderRequest, value, amount, supplier)).orElseGet(() -> isNotPresent(orderRequest));
```

Comment:
Jak tu zmienisz parametr z Optionala na zwykły obiekt, to nie będziesz w tym miejscu miał problemu.